src/services/database.ts

```diff
@@ -93,3 +107,52 @@ const CriteriaIndexMap: Record<Exclude<SearchCriteria, SearchCriteria.ALL>, Tabl
   vin: 'VinIndex',
   trailerId: 'TrailerIdIndex',
 };
+export const postTechRecord = async (request: any) => {
+  const systemNumber = await generateSystemNumber();
+  const { vin } = request;
+  request.systemNumber = systemNumber;
+  request.createdTimestamp = new Date().toISOString();
+  // eslint-disable-next-line @typescript-eslint/no-unsafe-call
+  request.partialVin = vin.length < 6 ? vin : vin.substring(request.vin.length - 6);
+  const params = {
+    TableName: tableName,
+    Item: request,
+    ConditionExpression: '#vin <> :vin AND #systemNumber <> :systemNumber',
+    ExpressionAttributeNames: {
+      '#vin': 'vin',
+      '#systemNumber': 'systemNumber',
+    },
+    ExpressionAttributeValues: {
+      ':vin': { S: vin },
+      ':systemNumber': { S: systemNumber },
+    },
+  };
+  return ddbClient.send(new PutItemCommand(params)).catch((x) => logger.error(x));
+};
+
+export const generateSystemNumber = async () : Promise<string> => {
+  try {
+    if (process.env.AWS_SAM_LOCAL) {
+      return '123';
+    }
+    const input = {
+      path: '/system-number/',
+      httpMethod: 'POST',
+      resource: '/system-number/',
+    };
+
+    const command = new InvokeCommand({
+      FunctionName: process.env.TEST_NUMBER_LAMBDA_NAME,
+      InvocationType: 'RequestResponse',
+      Payload: JSON.stringify(input),
+    });
+
+    const response = await lambdaClient.send(command);
+    const bufferResponse = Buffer.from(response.Payload!).toString('utf-8');
+    const bufferBody = JSON.parse(bufferResponse).body;
+    return JSON.parse(bufferBody).systemNumber;
+  } catch (e) {
+    logger.error(`Error in generate system number ${JSON.stringify(e)}`);
+    throw new Error('lambda client failed getting data');
+  }
+};
```

Comment:
Probably best to put the number generation logic in a separate file. 
Additionally, we need to be able to generate TrailerIds, ZNumbers and TNumbers. 

If a vehicle is posted without a VRM, we need to set the primaryVrm to:
- For a car, a new ZNumber
- For a HGV, a new ZNumber
- For a LGV, a new ZNumber
- For a motorcyle, a new ZNumber
- For a PSV, a new ZNumber

For trailers, if it does not exist, we need to set the trailerId to:
- For a Small Trailer, a new TNumber
- For a Trailer, a new TrailerId (CNumber)